compiler/rustc_target/src/spec/mod.rs

```diff
@@ -2063,7 +2064,8 @@ pub struct TargetOptions {
     /// However, parts of the backend do check this field for specific values to enable special behavior.
     pub abi: StaticCow<str>,
     /// Vendor name to use for conditional compilation (`target_vendor`). Defaults to "unknown".
-    pub vendor: StaticCow<str>,
+    #[rustc_lint_opt_deny_field_access("use `Target::is_like_*` instead of this field")]
```

Comment:
Maybe?

```suggestion
    #[rustc_lint_opt_deny_field_access("the vendor should ideally not affect compilation, use `Target::is_like_*` instead of this field")]
```

Reply:
Resolved by https://github.com/rust-lang/rust/pull/148760#discussion_r2508320242